code/modules/power/tesla/energy_ball.dm

```diff
@@ -321,16 +334,42 @@
 	else if(closest_mob)
 		var/shock_damage = (tesla_flags & TESLA_MOB_DAMAGE)? (min(round(power/600), 90) + rand(-5, 5)) : 0
 		closest_mob.electrocute_act(shock_damage, source, 1, tesla_shock = 1, stun = (tesla_flags & TESLA_MOB_STUN))
+		if(zap_gib && closest_mob.mob_siemens_coeff_amount > 0)
+			//no gib() because if chest is gone it cannot transfer tesla to other mobs
+			for(var/obj/item/W in closest_mob)
+				closest_mob.dropItemToGround(W)
+				if(prob(50))
+					W.throw_at(get_edge_target_turf(closest_mob,pick(GLOB.alldirs)),rand(1,3),5)
+			ADD_TRAIT(closest_mob, TRAIT_DISFIGURED, TRAIT_GENERIC)
+			sleep(0.1 SECONDS)
```

Comment:
Using sleep is very bad, this should be a callback timer or something instead.

Reply:
sleep here doesnt needed anyway, i put it there so the zap can travel slowly from person to person

Reply:
i will just remove it anyway

Reply:
Up to you, sleep in code is just bad in general because it's a byond built-in feature we have no control over, while we do over timers and callbacks, so they can do things like respect master controller. It's also easy to break and/or cause harddels.

Reply:
i see